specification/purview/resource-manager/Microsoft.Purview/preview/2024-04-01-preview/purview.json

```diff
@@ -2616,7 +2697,8 @@
         "properties": {
           "$ref": "#/definitions/PrivateLinkResourceProperties",
           "description": "The private link resource properties.",
-          "readOnly": true
+          "readOnly": true,
```

Comment:
It seems like you're changing your private links related definitions. Would this be a good opportunity to refactor to using common-types privatelinks.json?

Any blockers to doing so that you've identified?

Reply:
I looked into the common-types and have a few questions that could potentially present as blockers. There are some fields that seem to misalign from what we store in our repository and the common-types definitions.
For example, common-types defines PrivateEndpointConnection as a Resource, but we represent it as a ProxyResource. Similarly, the group-ids field in PrivateEndpointConnectionProperties is not present in our repo (likely because we are only storing the information as a proxy and connection, not the full resource) so that also misaligns.

What is the advice here in terms of refactoring? My concern on switching to the common-types is we may lose this nuance in our system (e.g. ProxyResource) and have data that is never actually returned by our system since we do not store it anywhere (e.g. group-ids).

Reply:
I think technically `ProxyResource` should be correct, and they are both similar - both definitions should not include things that are specific to TrackedResource. I don't think switching between them is likely to cause any issues, and the automated breaking changes validations that run on PRs here should hopefully catch any issues.

I am not sure whether group-ids is one of those parts of the private link contract that is not universally the same between RPs. I'd suggest we try to clarify that with the owners of the contract.

As long as the property is optional, and not required, I think technically its fine not to return it - see, for instance the way nextLink is always declared, but not actually always in responses. Adding it to your documented properties may look like a breaking change, and we can run that past breaking change reviewers to see what they think about the issue, but if you never actually return the property, or never return it except for new api-versions, I don't see how adding it to your documented set of optional properties is going to ever break anybody in practice.

I think it would still be an SDK breaking change, but hopefully if so it is at least one with some small upsides - overall helping better structure and maintain SDKs - and you're wanting to make some changes to private links to fix quality issues in your definitions anyway right? Including one enum change: PrivateEndpointConnectionStatus

Would breaking change experts like to add more guidance on whether its better or not to refactor at this point?

/cc @mikekistler @JeffreyRichter

Reply:
PS my general understanding, is that group-id should be specified by your service since it used when creating private endpoint connection(s) to specify the logical 'group' of endpoints which this private endpoint connection is to connect to. For instance, some services might have multiple hosted-on-behalf FQDNs that the customer can use, to connect to the service, and choose to split those up into functional groups for specific purposes the customer wishes to enable, e.g. (data ingestion, service management, persistence, monitoring).

I can see that maybe your API seems to be fine without providing this kind of information, but I also feel that in principal all private endpoint connections should provide this information. And it shouldn't actually be a breaking change, to start returning it, when previously or not.

Reply:
I will add this into the task to also investigate the addition of group ids for private endpoints. Thanks for the information Tim!